src/equations/shallow_water_two_layer_1d.jl

```diff
@@ -0,0 +1,585 @@
+It should be noted that the equations are ordered differently and the
+designation of the upper and lower layer has been changed which leads to a slightly different
+formulation.
+"""
+@inline function flux_fjordholm_etal(u_ll, u_rr,
+                                      orientation::Integer,
+                                      equations::ShallowWaterTwoLayerEquations1D)
+  # Unpack left and right state
+  h1_ll, h2_ll = waterheight(u_ll, equations)
+  v1_ll, v2_ll = velocity(u_ll, equations)
+  h1_rr, h2_rr = waterheight(u_rr, equations)
+  v1_rr, v2_rr = velocity(u_rr, equations)
+
+  # Average each factor of products in flux
+  h1_avg = 0.5 * (h1_ll + h1_rr)
+  h2_avg = 0.5 * (h2_ll + h2_rr)
+  v1_avg = 0.5 * (v1_ll + v1_rr)
+  v2_avg = 0.5 * (v2_ll + v2_rr)
+  p1_avg = 0.25 * equations.gravity * (h1_ll^2 + h1_rr^2)
+  p2_avg = 0.25 * equations.gravity * (h2_ll^2 + h2_rr^2)
+
+  # Calculate fluxes depending on orientation
+  f1 = h1_avg * v1_avg
+  f2 = f1 * v1_avg + p1_avg
+  f3 = h2_avg * v2_avg
+  f4 = f3 * v2_avg + p2_avg
+
+  return SVector(f1, f2, f3, f4, zero(eltype(u_ll)))
+end
+
+
+"""
+    flux_wintermeyer_etal(u_ll, u_rr, orientation,
+                          equations::ShallowWaterTwoLayerEquations1D)
+
+!!! warning "Experimental code"
+This numerical flux is experimental and may change in any future release.
+                      
+Total energy conservative (mathematical entropy for two-layer shallow water equations) split form.
+When the bottom topography is nonzero this scheme will be well-balanced when used as a `volume_flux`.
+The `surface_flux` should still use, e.g., [`flux_fjordholm_etal`](@ref). To obtain the flux for the
+two-layer shallow water equations the flux that is described in the paper for the normal shallow 
+water equations is used within each layer.
+
+Further details are available in Theorem 1 of the paper:
+- Niklas Wintermeyer, Andrew R. Winters, Gregor J. Gassner and David A. Kopriva (2017)
+  An entropy stable nodal discontinuous Galerkin method for the two dimensional
+  shallow water equations on unstructured curvilinear meshes with discontinuous bathymetry
+  [DOI: 10.1016/j.jcp.2017.03.036](https://doi.org/10.1016/j.jcp.2017.03.036)
+"""
+#TODO: Change name after paper is finished
+@inline function flux_wintermeyer_etal(u_ll, u_rr,
+                                        orientation::Integer,
+                                        equations::ShallowWaterTwoLayerEquations1D)
+  # Unpack left and right state
+  h1_ll, h1_v1_ll, h2_ll, h2_v2_ll, _ = u_ll
+  h1_rr, h1_v1_rr, h2_rr, h2_v2_rr, _ = u_rr
+
+  # Get the velocities on either side
+  v1_ll, v2_ll = velocity(u_ll, equations)
+  v1_rr, v2_rr = velocity(u_rr, equations)
+
+  # Average each factor of products in flux
+  v1_avg = 0.5 * (v1_ll + v1_rr)
+  v2_avg = 0.5 * (v2_ll + v2_rr)
+  p1_avg = 0.5 * equations.gravity * h1_ll * h1_rr
+  p2_avg = 0.5 * equations.gravity * h2_ll * h2_rr
+
+  # Calculate fluxes depending on orientation
```

Comment:
```suggestion
  # Calculate fluxes
```
There is no orientation dependence on the 1D TreeMesh